src/ast/irbuilderbpf.cpp

```diff
@@ -264,71 +264,83 @@ void IRBuilderBPF::CreateMapUpdateElem(Map &map, AllocaInst *key, Value *val)
 
 void IRBuilderBPF::CreateMapDeleteElem(Map &map, AllocaInst *key)
 {
-  Value *map_ptr = CreateBpfPseudoCall(map);
+  Value *mapid = CreateBpfPseudoCall(map);
+
+  assert(key->getType()->isPointerTy());
+  assert(mapid->getType()->isIntegerTy());
 
   // int map_delete_elem(&map, &key)
   // Return: 0 on success or negative error
   FunctionType *delete_func_type = FunctionType::get(
       getInt64Ty(),
```

Comment:
Isn't `getInt32Ty()` appropriate here (and other helper functions returning int)?

Reply:
ah good catch. I'm not sure if it actually matters for bpf but   it won't hurt to fix it :)